File: cpp/step/stack/main_9012.cpp
#include <iostream>
using namespace std;

int main() {
	int count;
	cin >> count;
	for (int i = 0; i < count; ++i) {
		string line;
		cin >> line;

		int check = 0;
		for (auto && c : line) {
			if (c == '(') {
				check++;
			}
			if (c == ')') {
				check--;
			}
			if (check < 0) {
				break;
			}
		}
		if (check == 0) {
			cout << "YES" << '\n';
		} else {
			cout << "NO" << '\n';
		}
	}
}

File: kotlin/step/stack/main_9012.kt
fun main() {
    val count = readLine()!!.toInt()
    for (i in 1..count){
        var open = 0
        for (c in readLine()!!){
            when (c) {
                '(' -> open++
                else -> open--
            }

            if (open < 0)
                break
        }
        when (open) {
            0 -> println("YES")
            else -> println("NO")
        }
    }
}